02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,371 @@
+
+In this analysis, we will be using `hcop` to map between zebrafish and human identifiers [@Turner2018].
+
+We'll need to use the `devtools` package to install `hcop` from Github [@Wickham2020].
+
+```{r}
+if (!("devtools" %in% installed.packages())) {
+  install.packages("devtools")
+}
+```
+
+Now we can install `hcop` from Github.
+You can control what version of the package is installed by using the `ref` command of `devtools::install_github`.
+Here, we'll use the most recent commit at the time that we are putting together this example.
+
+```{r}
+devtools::install_github("stephenturner/hcop", ref = "0985fddc91a6ef2308f4800958dfd11c25fe6a98")
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(hcop)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE71270.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE71270.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+## Mapping zebrafish Ensembl gene IDs to human symbols
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the `Gene` column of `df`
+head(df$Gene)
+```
+
+Enembl gene identifiers have different species-specific prefixes. 
+In zebrafish, Ensembl gene identifiers begin with `ENSDARG` (in human, `ENSG`, etc.).
+
+Now let's do the conversion!
+We'll use the `zebrafish` data from `hcop`; this is what `zebrafish` looks like.
+
+```{r}
+zebrafish
+```
+
+We're interested in the `human_symbol` and `zebrafish_ensembl` columns specifically.
+
+```{r}
+human_zebrafish_key <- zebrafish %>%
+  # we'll want to subset zebrafish to only the columns we're interested in
+  dplyr::select(zebrafish_ensembl, human_symbol)
```

Comment:
Can make this its own chunk? I see it as being its own step.

Reply:
Also based on the output where we have duplicated values, this tells me that there are duplicated keys here (because remember `zebrafish` has more the `symbols` so they would repeat `zebrafish_ensembl` and  `human_symbol` for these instances. Because of this, we should make sure our key doesn't have replicates before we do a `join()`. Try this and let me know if that works. 
```suggestion
  dplyr::select(zebrafish_ensembl, human_symbol) %>% 
  dplyr::distinct() 
```
We will want to carefully explain what this `distinct()` and why it's necessary. May even want to take a minute to show the duplicated data before applying it.